pkg/dumper/postgres/dumper.go

```diff
@@ -23,119 +22,72 @@ type pgDumper struct {
 }
 
 func NewDumper(conn *sql.DB, rdr reader.Reader) dumper.Dumper {
-	return &pgDumper{
-		conn:   conn,
-		reader: rdr,
-	}
+	return generic.NewSqlDumper(
+		rdr,
+		&pgDumper{
+			conn:   conn,
+			reader: rdr,
+		},
+	)
 }
 
-func (p *pgDumper) Dump(done chan<- struct{}, configTables config.Tables) error {
-	if err := p.dumpStructure(); err != nil {
+func (p *pgDumper) DumpStructure(sql string) error {
+	if _, err := p.conn.Exec(sql); err != nil {
```

Comment:
Here `return p.conn.Exec(sql)` would be enough

Reply:
See https://github.com/hellofresh/klepto/pull/58#discussion_r160352049